tests/indexes/test_indexes.py

```diff
@@ -1601,3 +1601,22 @@ def test_to_series() -> None:
         np.complexfloating,
     )
     check(assert_type(Index(["1"]).to_series(), "pd.Series[str]"), pd.Series, str)
+
+
+def test_multiindex_swaplevel_rettype() -> None:
+    """Test that union returns MultiIndex on MultiIndex input and swaplevel returns Self"""
+    mi = pd.MultiIndex.from_product([["a", "b"], [1, 2]], names=["let", "num"])
+    mi2 = pd.MultiIndex.from_product([["a", "b"], [3, 4]], names=["let", "num"])
+
+    check(
+        assert_type(mi.swaplevel(0, 1), "pd.MultiIndex"),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(mi.union(mi2), "pd.MultiIndex"),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(mi.union([("c", 3), ("d", 4)]), "pd.MultiIndex"),
+        pd.MultiIndex,
+    )
```

Comment:
Please make two tests, since `swaplevel` and `union` are two independent functionalities:


```suggestion
def test_multiindex_union() -> None:
    """Test that MultiIndex.union returns MultiIndex"""
    mi = pd.MultiIndex.from_product([["a", "b"], [1, 2]], names=["let", "num"])
    mi2 = pd.MultiIndex.from_product([["a", "b"], [3, 4]], names=["let", "num"])

    check(
        assert_type(mi.union(mi2), "pd.MultiIndex"), pd.MultiIndex
    )
    check(
        assert_type(mi.union([("c", 3), ("d", 4)]), "pd.MultiIndex"), pd.MultiIndex
    )

    
def test_multiindex_swaplevel() -> None:
    """Test that MultiIndex.swaplevel returns MultiIndex"""
    mi = pd.MultiIndex.from_product([["a", "b"], [1, 2]], names=["let", "num"])
    check(
        assert_type(mi.swaplevel(0, 1), "pd.MultiIndex"), pd.MultiIndex
    )
```